spark_job_processor/schemas.py

```diff
@@ -0,0 +1,74 @@
+from collections import defaultdict
+from datetime import datetime
+from pydantic import BaseModel, Field
+
+
+
+
+class Model(BaseModel):
+
+    def __getitem__(self, item):
+        return self.__dict__[item]
+
+    def __setitem__(self, item, value):
+        self.__dict__[item] = value
+
+
+def get_total_model(models: list[Model]) -> Model:
+    max_keys = ['jvm_peak_memory', 'python_peak_memory', 'other_peak_memory', 'total_peak_memory']
+    total_model = models[0].copy()
```

Comment:
what if `len(models) == 0`?
I think it would be cleaner to pass `return_type` to the function (i.e SparkTask) and then:

```
def get_total_model(list, return_type)
    total_model = return_type()
    for model in models:
        ....
```